url9ba68db77a_samkio_Dungeonator_tgz-pJ8-net_woopa_dungeon_core_Main2J8/src/net/woopa/dungeon/core/RoomPopulator.java

```diff
@@ -0,0 +1,256 @@
+			for (int i = 0; i <= RandomUtil.nextInt(3); i++)
+				objectRandom(CoreMaterial.SHROOM);
+
+		if (RandomUtil.chance(SettingsManager.getInt(CoreSettings.BenchPct)))
+			objectRandom(CoreMaterial.WORKBENCH);
+
+		if (RandomUtil.chance(SettingsManager.getInt(CoreSettings.OvenPct)))
+			objectRandom(CoreMaterial.FURNACE);
+
+		if (RandomUtil.chance(SettingsManager.getInt(CoreSettings.AnvilPct)))
+			objectRandom(CoreMaterial.ANVIL);
+
+		if (!(room.getType().equals(RoomType.SPECIAL))
+				&& RandomUtil.chance(SettingsManager
+						.getInt(CoreSettings.PoolPct)))
+			poolRandom();
+
+		if (!(room.getType().equals(RoomType.SPECIAL))
+				&& RandomUtil.chance(SettingsManager
+						.getInt(CoreSettings.SandPct)))
+			allFloor(CoreMaterial.SOULSAND, 2);
+	}
+
+	private void allFloor(Material m, int inset) {
+		int px = room.getSizeX() - inset * 2;
+		int py = room.getSizeY() - inset * 2;
+
+		for (int x = 0; x < px; x++) {
+			for (int y = 0; y < py; y++) {
+				int px1 = room.getOriginX() + x + inset;
+				int py1 = room.getOriginY() + y + inset;
+				if (room.getGrid().get(px1, py1) == CoreMaterial.FLOOR) {
+					room.getGrid().set(px1, py1, m);
+				}
+			}
+		}
+	}
+
+/*	private Boolean trapRandom() {
+		wallLoc loc;
+		if (true) {
+			loc = wallLocation(this);
+			if (loc != null) {
+				int dx = loc.dir.rotate180().dx(1);
+				int dy = loc.dir.rotate180().dy(1);
+				if (room.getGrid().get(loc.x, loc.y) == CoreMaterial.WALL
+						&& room.getGrid().get(loc.x + dx, loc.y + dy) == CoreMaterial.FLOOR
+						&& room.getGrid().get(loc.x + dx * 2, loc.y + dy * 2) == CoreMaterial.FLOOR
+						&& room.getGrid().get(loc.x + dx * 3, loc.y + dy * 3) == CoreMaterial.FLOOR) {
+					room.getGrid().set(loc.x, loc.y, CoreMaterial.ARROW);
+					room.getGrid().set(loc.x + dx, loc.y + dy, CoreMaterial.RED2);
+					room.getGrid().set(loc.x + dx * 2, loc.y + dy * 2, CoreMaterial.RED1);
+					room.getGrid().set(loc.x + dx * 3, loc.y + dy * 3,
+							CoreMaterial.PRESSURE);
+					return true;
+				}
+			}
+		}
+		return false;
+	}*/
+
+	private Boolean chestRandom() {
+		int inset = 1;
+
+		int px = room.getSizeX() - inset * 2;
+		int py = room.getSizeY() - inset * 2;
+
+		if (px > 0 && py > 0 && (room.getSizeX() > 3 && room.getSizeY() > 3)) {
+			int px1, py1;
+			for (int at = 0; at < 100; at++) {
+				px1 = room.getOriginX() + RandomUtil.nextInt(px) + inset;
+				py1 = room.getOriginY() + RandomUtil.nextInt(py) + inset;
+				if (okForChest(px1, py1, room.getGrid())) {
+					room.getGrid().set(px1, py1, CoreMaterial.CHEST);
+					return true;
+				}
+			}
+		}
+		return false;
+	}
+
+	private boolean okForChest(int x, int y, Grid g) {
+		if (g.get(x, y) != CoreMaterial.FLOOR) // Must be plain floor (not fixed
+												// foor etc)
+			return false;
+		if (g.isChest(x + 1, y) || g.isChest(x - 1, y) || g.isChest(x, y + 1)
+				|| g.isChest(x, y - 1))
+			return false;
+		return true;
+	}
+
+	private Boolean enchantRandom() {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 21. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.